design/rich-authorization-requests.md

```diff
@@ -0,0 +1,196 @@
+as granted by the resource owner and assigned to the respective
+access token.
+
+## Implementation details
+
+SPI
+
+````
+public interface RarProcessorProvider extends Provider {}
+
+public interface RarProcessorProviderFactory extends ProviderFactory<RarProcessorProvider> { }
+````
+e.g. implementation in extension
+
+````
+public class Psd2RarProcessorProvider implements RarProcessorProvider {}
+
+public class Psd2RarProcessorProviderFactory implements RarProcessorProviderFactory { 
+
+    public static final String ID = "rar-psd2";
+    public static final String SUPPORTED_DATA_TYPE = []; //Json array schema definition
+}
+````
+
+* Support advertisement of supported authorization details types in OAuth server metadata
+
+````
+OIDCConfigurationRepresentation config = new OIDCConfigurationRepresentation();
+config.setAuthorizationDataTypesSupported(
+    getSupportedAuthorizationDataTypes(RarProcessorProvider.class)
+    
+);
+````
+
+*  Check if authorization data type or authorization details is not conforming to the respective type
+   definition when authorization_details is available in the authorization requests 
+
+````
+rarProcessorProvider.isAuthorizationDetailsCompliant(authorizationDetails)
+````
+After the user authenticate, rarProcessor should:
+
+*  Determine how authorization details are shown to the user in the user consent
+*  Exchange with the resource server if necessary
+*  Enrich authorization details in the user consent process (e.g. add selected accounts or set expirations)
+*  show the consent page
+
+````
+rarProcessorProvider.prepareAuthorizationDetails()
+````
+   
+When the user submits the consent  
+
+*  Merge requested and pre-existing authorization_details (e.g. in case of update)  
+
+* store authorization_details within the grant
+
+````
+rarProcessorProvider.processAuthorizationDetails()
+````
+
+### Admin UI
+
+AS should allow the possibility to select RarProcessor Implementation Module.
+
+Files/Classes/methods affected:
+
+### Adapter
+
+Some adapters should be updated in order to support the new authorization_details
+
+Files/Classes/methods affected:
+
+* Keycloak adapter-core (org.keycloak.adapters.OAuthRequestAuthenticator#authenticate)
+
+* Keycloak JS Adapter (loginUrl creation with authorization_details parameter [PoC for Keycloak.js](https://gist.github.com/thomasdarimont/3e87944c31b6263f1849e35733a03500#file-index-html-L204)) 
+  
+* Access to authorization_details in tokenResponse see [keycloak.js](https://github.com/keycloak/keycloak/blob/b1d39aa136662d54deca10c97374f0932ce6316b/adapters/oidc/js/src/main/resources/keycloak.js#L675).
+
+
+
+
+## Tests
+RAR should be properly covered by unit and integration tests.
+
+## Documentation
+RAR usage should be properly documented.
+
+Affected documents: Securing Applications and Services Guide
+
+## Resources
+* [draft-ietf-oauth-rar][1]
+
+[1]: https://tools.ietf.org/html/draft-ietf-oauth-rar-04
```

Comment:
FYI. Just a minor note that newest draft is https://datatracker.ietf.org/doc/html/draft-ietf-oauth-rar-07 ...

Reply:
yes, I looked at the difference, it's more some text. The concept didn't change and no new parameter.